crates/bonsaidb-core/src/test_util.rs

```diff
@@ -881,6 +881,9 @@ pub async fn list_tests<C: Connection>(db: &C) -> anyhow::Result<()> {
     assert_eq!(both_docs[0].contents.value, doc1_value.value);
     assert_eq!(both_docs[1].contents.value, doc2_value.value);
 
+    // Test list_headers() method
+    let both_headers = collection.list_headers(doc1.id..=doc2.id, db).await?;
```

Comment:
The headers should be compared against the known headers (`doc1`/`doc2`).

Reply:
Yeah that part is remaining. Currently getting following error.

```
vbmade2000@vbmade2000:/mnt/ubuntu_extra/projects/rust/bonsaidb$ cargo test --all-features --package bonsaidb-local
   Compiling bonsaidb-core v0.3.0 (/mnt/ubuntu_extra/projects/rust/bonsaidb/crates/bonsaidb-core)
error[E0698]: type inside `async fn` body must be known in this context
   --> crates/bonsaidb-core/src/test_util.rs:891:10
    |
891 |         .list_headers(doc1.id..=doc2.id, Sort::Ascending, None)
    |          ^^^^^^^^^^^^ cannot infer type for type parameter `C` declared on the associated function `list_headers`
    |
note: the type is part of the `async fn` body because of this `await`
   --> crates/bonsaidb-core/src/test_util.rs:891:64
    |
891 |           .list_headers(doc1.id..=doc2.id, Sort::Ascending, None)
    |  ________________________________________________________________^
892 | |         .await?;
    | |______________^

For more information about this error, try `rustc --explain E0698`.

```